clientv3/client.go

```diff
@@ -295,7 +295,13 @@ func isHaltErr(ctx context.Context, err error) bool {
 	if err == nil {
 		return false
 	}
-	return grpc.Code(err) != codes.Unavailable
+	code := grpc.Code(err)
+	// Unavailable codes mean the system will be right back.
+	// (e.g., can't connect, lost leader)
+	// Internal codes seem to mean something failed, leaving the
+	// system in an inconsistent state, but retrying could make progress.
+	// (e.g., failed in middle of send, corrupted frame)
+	return code != codes.Unavailable && code != codes.Internal
```

Comment:
probably add a todo here. i do not really like the work `seem to mean` here. Eventually, we should have confidence on error handling.